lib/bolt/transport/api.rb

```diff
@@ -7,21 +7,27 @@
 module Bolt
   module Transport
     class Api < Orch
-      def initialize(*args)
-        super
+      attr_reader :plan_job
+
+      def initialize(plan_job, *args)
+        @plan_job = plan_job
+        @connection = nil
+        super(*args)
       end
 
       def get_connection(conn_opts)
-        key = Bolt::Transport::Api::Connection.get_key(conn_opts)
-        unless (conn = @connections[key])
-          @connections[key] = Bolt::Transport::Api::Connection.new(conn_opts, logger)
-          conn = @connections[key]
-        end
-        conn
+        @connection ||= Bolt::Transport::Api::Connection.new(conn_opts, logger, @plan_job)
       end
 
       def batches(targets)
-        targets.group_by { |target| Bolt::Transport::Api::Connection.get_key(target.options) }.values
+        targets
+      end
+
+      def finish_plan(result)
+        # CODEREVIEW should we check that result is a Bolt::PlanResult?
```

Comment:
That does not seem necessary since it should be coming from our code.

TODO: Verify that the plan result is generated by whatever call this.